### PR TITLE
toggle favorites

### PR DESCRIPTION
**Pull request by williambaibai**
changed `user` model an added dao methods and controller for favoriting

### CODE REVIEW

File: src/app/models/user.py [hunk 1]
```diff
@@ -1,16 +1,27 @@
 from . import *
 
+# define many-to-many relationship
+users_to_gymclasses = db.Table('user_to_gymclasses',
```
Comment (young-k) [inline src/app/models/user.py]: can we make this:
```
users_to_gymclasses = db.Table(
    'user_to_gymclasses',
    ...
)

File: src/app/controllers/toggle_favorite.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+from . import *
+
+class ToggleFavoriteController(AppDevController):
+
+  def get_path(self):
+    return '/favorite/<device_id>/<gymclass_id>/'
+
+  def get_methods(self):
+    return ['POST', 'GET']
```
Comment (young-k) [inline src/app/controllers/toggle_favorite.py]: The route should be `'/favorite/<gymclass_id>`, and the `device_id` should be sent in the POST body

Why is it POST & GET though? Shouldn't it just be POST?

File: src/app/dao/users_dao.py [hunk 3]
```diff
@@ -3,24 +3,29 @@
...
+def is_registered(device_id):
+  return get_user_by_device_id(device_id) is not None
 
-def create_user(email, first_name='', last_name='', image_url=''):
-  optional_user = get_user_by_email(email)
+def get_user_classes(user_id):
+  optional_user = get_user_by_id(user_id)
+  if optional_user is None:
+    raise Exception('User does not exist.')
+  return optional_user.gym_classess
```
Comment (young-k) [inline src/app/dao/users_dao.py]: Typo, should be `gym_classes`

File: src/app/dao/users_dao.py [hunk 4]
```diff
@@ -3,24 +3,29 @@
...
+
+def create_user(device_id, first_name='', last_name=''):
```
Comment (young-k) [inline src/app/dao/users_dao.py]: Can we get rid of the `first_name` and `last_name` as well? We can always add these in later

File: src/app/controllers/toggle_favorite.py (hunk 2 above)
Comment (williambaibai) [reply src/app/controllers/toggle_favorite.py]: ok, how do we get `device_id` from the POST body?

Comment (young-k) [reply src/app/controllers/toggle_favorite.py]: https://stackoverflow.com/questions/10434599/how-to-get-data-received-in-flask-request

**Pull request merged**